### PR TITLE
Typescript errors and Implement Router

### PR DESCRIPTION
**Pull request by brabbuss**
##### Resolves several errors and implement Router:
- 1 - changed proxy server for CORS workaround to allow a higher rate limit
- 2 - Implement Switch/Router/Link and add proper endpoints
- 3 - Implement addFavorite() to add songs to favorites list
- 4 - create file to store and export interfaces/types
- 5 - resolve numerous TS errors halting production

#### Where should the reviewer start?
- opening the app, user should be able to see results at the '/results' endpoint upon clicking submit
- note  issue  #20  where some choices return nothing and will cause an error

### CODE REVIEW

File: src/components/App/App.tsx [hunk 1]
```diff
@@ -1,38 +1,46 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useEffect } from "react";
```
Comment (demaceo) [inline src/components/App/App.tsx:1]: Do we need useEffect? I don't see it being called anywhere

**Review by demaceo — APPROVED** [review]
This is fantastic. Relieved some of those bugs were managed

File: src/components/App/App.tsx [hunk 2]
```diff
@@ -1,38 +1,46 @@
...
+      .filter(musicGenre => genre !== musicGenre)
+      .join(",");
+    const results = await getTracksByMoodAPI(valence, arousal, excludedGenres, decade)
+    setSongResults(results);
   };
 
+  const addFavorite: Function = (id: string) => {
+    type AnyType = any;
+    const favorite: ISongResults = songResults.find((song:ISongResults) => song.id === id) as AnyType // favorite needs to be set to any???
+    setFavoriteSongs([favorite, ...favoriteSongs]) // putting these params inside an array (error expected 1 arg but got 2+)
```
Comment (bruce-gordon) [inline src/components/App/App.tsx:28]: Great job figuring this out!  I don't totally understand "AnyType".  Let's go over this today!

File: src/components/common/Types.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
+export const allGenres: string[] = [
```
Comment (bruce-gordon) [inline src/components/common/Types.ts:1]: Good organization here.  This makes everything else look cleaner.

**Review by bruce-gordon — APPROVED** [review]

**Pull request merged**